engines/director/lingo/lingo-builtins.cpp

```diff
@@ -2033,50 +2033,49 @@ void LB::b_installMenu(int nargs) {
 
 	menu->setCommandsCallback(menuCommandsCallback, g_director);
 
-	debugC(3, kDebugLingoExec, "installMenu: '%s'", Common::toPrintable(menuStxt).encode().c_str());
+	debugC(3, kDebugLingoExec, "installMenu: '%s'", Common::toPrintable(menuStxt).c_str());
 
 	LingoArchive *mainArchive = movie->getMainLingoArch();
 
-	// Since loading the STXT converts the text to Unicode based on the platform
-	// encoding, we need to fetch the correct Unicode character for the platform.
-
 	// STXT sections use Mac-style carriage returns for line breaks.
-	// The code preprocessor converts carriage returns to line feeds.
-	const uint32 LINE_BREAK = 0x0a;
-	// Menu definitions use the character 0xc3 to denote a checkmark.
-	// For Mac, this is √. For Windows, this is Ã.
-	const uint8 CHECKMARK_CHAR = 0xc3;
-	const uint32 CHECKMARK_U32 = numToChar(CHECKMARK_CHAR);
-	const char *CHECKMARK_STR = "\xc3\x83"; // "Ã"
+	const char LINE_BREAK_CHAR = 13;
 	// Menu definitions use the character 0xc5 to denote a code separator.
 	// For Mac, this is ≈. For Windows, this is Å.
-	const uint8 CODE_SEPARATOR_CHAR = 0xc5;
-	const uint32 CODE_SEPARATOR_U32 = numToChar(CODE_SEPARATOR_CHAR);
-	const char *CODE_SEPARATOR_STR = "\xc3\x85"; // "Å"
-
-	Common::U32String lineBuffer;
-
-	for (const Common::u32char_type_t *s = menuStxt.c_str(); *s; s++) {
-		lineBuffer.clear();
-		while (*s && *s != LINE_BREAK) {
-			if (*s == CHECKMARK_U32) {
-				lineBuffer += CHECKMARK_CHAR;
-				s++;
-			} else if (*s == CODE_SEPARATOR_U32) {
-				lineBuffer += CODE_SEPARATOR_CHAR;
-				s++;
-			} else if (*s == CONTINUATION) { // fast forward to the next line
-				s++;
-				if (*s == LINE_BREAK) {
-					lineBuffer += ' ';
-					s++;
+	const char CODE_SEPARATOR_CHAR = -59;
```

Comment:
This is problematic. Please use `byte` instead, which is unsigned. The thing is that `char` could be signed or unsigned depending on the platform and having a negative here could lead to portability problems.

The most portable and dev-friendly version is to use hex constants `0xC5` or the hex notation like `\xC5`

Reply:
Thanks, casting unsigned 0xc5 to signed raises compiler errors and 2am brain forgot about `\xC5'